ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/KsqlMaterialization.java

```diff
@@ -185,16 +185,38 @@ public MaterializedWindowedTable windowed() {
     public Iterator<WindowedRow> get(final int partition, final Range<Instant> windowStartBounds,
         final Range<Instant> windowEndBounds) {
       final Iterator<WindowedRow> result = table.get(partition, windowStartBounds, windowEndBounds);
-
       return Streams.stream(result)
           .map(row ->  {
-            return filterAndTransform(row.windowedKey(), row.value(), row.rowTime())
+            return filterAndTransform(row.windowedKey(), getIntermediateRow(row), row.rowTime())
                 .map(v -> row.withValue(v, schema()));
           })
           .filter(Optional::isPresent)
           .map(Optional::get)
           .iterator();
     }
   }
+
+  public static GenericRow getIntermediateRow(final TableRow row) {
```

Comment:
This method can be broken down into 2 methods: One for Regular rows, and the other for Windowed rows. Thoughts?

Reply:
I don't think it's a big deal since it can be figured out by the row itself.  They also share other fields you'd have to otherwise duplicate.